tests/features/user/masquerade.feature

```diff
@@ -0,0 +1,102 @@
+@api
+Feature:
+  - As a moderator, granted with 'edit masquerade field' permission, in order to
+    allow several users to masquerade, I am able to edit the 'Masquerade as'
+    field on their profile edit form and set accounts that this user is able to
+    masquerade as.
+  - As moderator, I can see the the 'Masquerade as' link list on a user profile
+    page but clicking on any link will redirect me to the target user profile.
+  - As a user able to masquerade, when I click on a target account link in my
+    profile page, I'm masquerading as that user. If that user can masquerade, I
+    cannot do it once more as I'm already masquerading.
+
+  Scenario: Test access on 'Masquerade as' field in the profile edit form.
+
+    Given users:
+      | Username             | First name | Family name |
+      | i_want_to_masquerade | Woz        | Dizzy       |
+      | target_account_1     |            |             |
+      | target_account_2     |            |             |
+    And I am logged in as a moderator
+
+    When I visit the profile of i_want_to_masquerade
+    And I click Edit
+    Then the following fields should be present "Masquerade as"
+
+    When I fill in "masquerade_as[0][target_id]" with "target_account_1"
+    And I press "Save"
+    And I click Edit
```

Comment:
I found a vulnerability here, a moderator can edit their own account and enter `joinup-admin` as a masquerade target and elevate their permission level this way. Normally the admin user is blocked on production but there are times when they are temporarily enabled. Also a user could possibly masquerade as a moderator, which would also exceed their permission level.

I think we should implement `hook_masquerade_access()` and limit the way people can exceed their own permission levels. The simplest way would be to only masquerade as users that are authenticated but have no other roles.

edit: This was discussed during standup, and it should be possible to masquerade as moderators, but not as administrator or super user.

Reply:
Great catch! As we've been discussed in the today's standup, this will be a follow-up in the _Masquerade_ epic. I've been opened ISAICP-5812.